### DISCUSSION TITLE
[PostgreSQL] Add an option to optimise the table size by sorting the columns

### DISCUSSION THREAD

**Discussion by tukusejssirs** [mikro-orm/mikro-orm#2849] [Ideas]
I have recently read about how PostgreSQL aligns the columns as data ([src](https://www.2ndquadrant.com/en/blog/on-rocks-and-sand/)). It might be a useful option to add to MikroORM. I don’t think it should be on by default, as some people prefer to define the column order by hand.

I suggest to at least order the columns by their byte length (see [this](https://www.2ndquadrant.com/en/blog/on-rocks-and-sand/#column-tetris), ideally, it might also be a good idea to try to fit the columns even more by trying to align the columns as much as possible.

**Comment by mjpowersjr** [comment]
This would be useful for me as well, although my use case different. I was attempting to integrate an existing database UI (nocodb) with my Mikro-ORM managed database - and hit a minor bug that could be worked around if I could influence the column positions.

**Reply by tukusejssirs** [reply]
Well, while I didn’t explicitly state it in the OP, I’d like to have an option in MikroORM which when I set it to `true`, it would _automatically_ re-order the columns in a table before the table creation.

Now, as I understand it, you want to _manually_ order the table columns, which is (AFAIK) how MikroORM currently works: it generates a `create table` query with the columns in the order you defined in the entity.

That said, you might wanted to change the column order once the table was created: AFAIK in PostgreSQL, it is not possible unless you re-create the table, as the column order is defined at table creation.
